src/tools/isabelle_executor.py

```diff
@@ -437,26 +453,35 @@ def _run_stmt_on_isabelle_server(self, stmt: str) -> None:
 
         # In proof mode. Execute tactics
         if self._proof_running or begin_clause:
-            stmt = stmt.strip()
+            stmt = f"{self.buffer}{stmt.strip()}"
 
-            # Run statement, with timeout. 
+            # Throw an error if "sorry" is used, which is not allowed
+            # Note that this is a fairly simple/optimistic way of handling it
+            if proof_search_mode and "sorry" in stmt:
+                raise Exception('Error: expected tactic, got "sorry". Do not use "sorry" in your proof.')
```

Comment:
We can use the same idea to prohibit sorry — when in proof search mode, sorry is not allowed as a tactic. Note that currently I check if the tactic contains the string sorry. In the rare case that a lemma has the string sorry in it, this won’t work.